Ginger/Ginger/AutomatePageLib/NewAutomatePage.xaml.cs

```diff
@@ -1091,18 +1096,73 @@ private void xSaveBusinessFlowBtn_Click(object sender, RoutedEventArgs e)
 
             Reporter.ToStatus(eStatusMsgKey.SaveItem, null, mBusinessFlow.Name,
                                       GingerDicser.GetTermResValue(eTermResKey.BusinessFlow));
-            WorkSpace.Instance.SolutionRepository.SaveRepositoryItem(mBusinessFlow);
+
+            SwapLoadingPrefixText("Saving", false);
+            xItemsTabsSection.Visibility = Visibility.Collapsed;
+            xItemsLoadingPnl.Visibility = Visibility.Visible;
+
+            await WorkSpace.Instance.SolutionRepository.SaveRepositoryItemAsync(mBusinessFlow);
+
             Reporter.HideStatusMessage();
+
+            xItemsTabsSection.Visibility = Visibility.Visible;
+            xItemsLoadingPnl.Visibility = Visibility.Collapsed;
+            SwapLoadingPrefixText("Saving", true);
+        }
+
+        public string loadingText { get; set; } = "Loading [BusinessFlow]...";
+
+        void SwapLoadingPrefixText(string swappedText, bool IsReset)
+        {
+            if (string.IsNullOrEmpty(swappedText))
+            {
+                return;
+            }
+
+            if (IsReset)
+            {
+                loadingText = loadingText.Replace(swappedText, "Loading");
+            }
+            else
+            {
+                loadingText = loadingText.Replace("Loading", swappedText);
+            }
+
+            xLoadWindowText.Text = loadingText;
         }
 
-        private void xUndoChangesBtn_Click(object sender, RoutedEventArgs e)
+        private async void xUndoChangesBtn_Click(object sender, RoutedEventArgs e)
         {
             if (CheckIfExecutionIsInProgress()) return;
 
-            if (Ginger.General.UndoChangesInRepositoryItem(mBusinessFlow, true))
+            if (Reporter.ToUser(eUserMsgKey.AskIfToUndoItemChanges, mBusinessFlow.ItemName) == eUserMsgSelection.Yes)
             {
-                mActivitiesPage.ListView.UpdateGrouping();
-                mBusinessFlow.SaveBackup();
+                xItemsTabsSection.Visibility = Visibility.Collapsed;
+                xItemsLoadingPnl.Visibility = Visibility.Visible;
+                SwapLoadingPrefixText("Undoing", false);
+
+                try
+                {
+                    //Mouse.OverrideCursor = System.Windows.Input.Cursors.Wait;
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5936174)